kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue

```diff
@@ -315,27 +297,30 @@
               ids: lesson.resources.map(resource => resource.contentnode_id),
             },
           }).then(contentNodes => {
-            return this.progressPromise(contentNodes);
+            const removeSelfFilter = node => node.id !== this.content.id;
+            if (this.$store.getters.isUserLoggedIn) {
+              this.mapSidePanelContentProgress(contentNodes).then(
+                () => (this.viewResourcesContents = contentNodes.filter(removeSelfFilter))
```

Comment:
I would suggest treating the contentNodeProgress load as a lazy load (this is how we treat it in other places in the codebase), and not block loading of the main content for it.

If you just rely on the built in ProgressBar functionality as I suggest above, it will reactively update when the progress data has been loaded.